tests/BootstrapperTest.php

```diff
@@ -184,6 +186,33 @@
     expect($new_storage_path)->toEqual($expected_storage_path);
 });
 
+test('tenant storage gets deleted after the tenant if filesystem.delete_storage_after_tenant_deletion is true', function () {
+    config([
+        'tenancy.bootstrappers' => [
+            FilesystemTenancyBootstrapper::class,
+        ],
+        'tenancy.filesystem.delete_storage_after_tenant_deletion' => true
+    ]);
+
+    Event::listen(TenantDeleted::class, function(TenantDeleted $event) {
+        if (config('tenancy.filesystem.delete_storage_after_tenant_deletion')) {
+            File::deleteDirectory($event->tenant->run(fn () => storage_path()));
+        }
+    });
+
+    tenancy()->initialize(Tenant::create());
+
+    Storage::fake('test');
+
+    expect(File::isDirectory(storage_path()))->toBeTrue();
+
+    Storage::put('test.txt', 'testing file');
+
+    tenant()->delete();
+
+    expect(File::isDirectory(storage_path()))->toBeFalse();
```

Comment:
It feels strange to use `storage_path()` here because it depends on the tenant's environment, but the tenant got already deleted.

So while the code is not wrong, it points out that tenancy remains initialized for a tenant even after the tenant is deleted.

To make the code a bit more clear it'd be best to fetch the tenant's `storage_path()` after initialization, save it into a variable, and then make assertions with that variable (i.e. replace your current `storage_path()` references), rather than fetching it. Because calling `storage_path()` each time doesn't make it 100% clear that it's always the same value.

```suggestion
    tenancy()->initialize(Tenant::create());
    $tenantStoragePath = storage_path();

    Storage::fake('test');

    expect(File::isDirectory($tenantStoragePath))->toBeTrue();

    Storage::put('test.txt', 'testing file');

    tenant()->delete();

    expect(File::isDirectory($tenantStoragePath))->toBeFalse();
```

Reply:
Right, this makes more sense